src/@types.ts

```diff
@@ -1,20 +1,63 @@
-export interface RawSchema {
+import { Resolver, ForSchemaOptions } from 'avsc'
+
+export enum SchemaType {
+  AVRO = 'AVRO',
+  JSON = 'JSON',
+  PROTOBUF = 'PROTOBUF',
+  UNKNOWN = 'UNKNOWN',
+}
+
+export interface SchemaHelper {
+  validate(schema: Schema): void
+  getSubject(confluentSchema: ConfluentSchema, schema: Schema, separator: string): ConfluentSubject
+}
+
+export type AvroOptions = Partial<ForSchemaOptions>
+export type JsonOptions = any // FIXME:
+export type ProtoOptions = any // FIXME:
+
+export interface LegacyOptions {
+  forSchemaOptions?: AvroOptions
+}
+export interface ProtocolOptions {
+  [SchemaType.AVRO]?: AvroOptions
+  [SchemaType.JSON]?: JsonOptions
+  [SchemaType.PROTOBUF]?: ProtoOptions
+}
+export type SchemaRegistryAPIClientOptions = ProtocolOptions | LegacyOptions
+
+export interface Schema {
+  toBuffer(payload: object): Buffer // FIXME:
```

Comment:
This one says FIXME, but I'm not quite sure what there is to fix here 😅  I'm not sure how we can constrain the type any further.

Reply:
I think it was copy-pasted from `master`. If it doesn't need fixing I can remove the comment